### DISCUSSION TITLE
Next version ...

### DISCUSSION THREAD

**Discussion by nunobrum** [nunobrum/spicelib#103] [Announcements]
## Next Version will...

This discussion aims to be a forum to decide what should integrate the next version of spicelib and what still needs improvements.

**Comment by nunobrum** [comment]
Given the large number of fixes and pull-requests, I suggest next version will be 1.3.0.
The following items have been corrected:

- Support for LTspice log files with the option : expanded netlist
- Supporting library symbols using BLOCK primitive
- Improved unittest on the .ASC hierarchical design
- SimRunner simulation iterator only returns successful simulations in order to simplify error management
- In QschEditor, the replacement of unique dot instructions (ex: .TRAN .AC .NOISE) is only done if the existing instruction is not commented.
- Fix: prefix was case sensitive
- Fix: Parsing netlists with extensions other than .net
- Fix: RunTask.get_results() returns None if a callback function is provided and the simulation has failed.

**Reply by hb020** [reply]
I'll also correct the documentation and content of log\semi_dev_op_reader.exe and qspicelogreader: it is incomplete.

**Reply by hb020** [reply]
adapted doc in the 'update-doc-for-1.3.0' branch.

* please see if the readme doc needs more (by the way, readme will now be included in readthedocs)
* the changelist for 1.3.0 is not in there yet

**Reply by nunobrum** [reply]
Just quickly browsed through `update-doc-for-1.3.0`. Great work there. I'll make a deeper dive into it during the weekend.

**Comment by nunobrum** [comment]
Hello @hb020 

I'm having a few problems generating the documentation. First compain was the missing myst_parser library.
After installing it using PiP, I now have a lot of error messages like this:

`C:\sandbox\spicelib_dev\README.md:: WARNING: 'myst' cross-reference target not found: 'what-is-contained-in-this-repository' [myst.xref_missing]`

What am I missing?

**Reply by nunobrum** [reply]
BTW, I'm using  myst_parser-3.0.1 in an old Python 3.8 installation.  Which build environment do you have ?

**Reply by hb020** [reply]
Using 3.12 as a base. 
Used myst-parser==4.0.0. I added it to conf.py. 
Don't know what python version readthedocs uses. Will see if I can fix it to a specific version that works from 3.8 to 3.12

**Reply by nunobrum** [reply]
I'll check that.

**Reply by hb020** [reply]
Have succeeded in forcing both 3.8 and 3.12 on sphinx (myst-parser 3.0.1 and 4.0.0), but now even 3.12 fails. Will need to do some digging tomorrow (shutting down for tonight)

Edit: both complain, but in both cases the end result is OK. No idea why it complains.

**Reply by nunobrum** [reply]
OK! I'm also a bit busted. I've installed 3.12 but having the same problem.

**Reply by hb020** [reply]
correction pushed. No more warnings. Works in both 3.8 and 3.12.

**Reply by nunobrum** [reply]
Great! I've complemented information on the qspice_log_reader. I won't touch the semi_dev_op_reader. There is no much else to say there and I guess not many people are using it anyway.
If there is nothing else from your side, I think we can create the wheel package for PyPi.

**Reply by hb020** [reply]
ok with me.

**Comment by hb020** [comment]
There's a nice message on readthedocs:

```
This is the latest development version
Some features may not yet be available in the published stable version. Read the [stable version of this documentation](https://spicelib.readthedocs.io/en/stable/).
```

Maybe because the 1.3.0 is not mentioned as release on github (1.2.1 is)? No hurry, just a heads up in case this was unintentional.

**Reply by nunobrum** [reply]
I must check what it is later. In any case it is non international.
